M3-4023 Screen readers cannot access "Create Linode" interface

WilkinsKa1 opened the pull request:
## Description

Our Add New Entity Menu was not accessible by keyboard. This presents obvious usability issues, particularly for users who use the keyboard as their primary (sometimes only) tool for navigation. In this PR we are bringing in Reach's MenuButton component to replace MUI's menu components that were originally in place. 

While we do lose some of our stylized transitional effects, these are not nearly as important as basic functionality (like keyboard access) for such a crucial interaction within the app. We can always add something similar in down the road, but for now I think this should be fixed sooner than later.

Additionally, I fixed our icons hovers.

## Type of Change
- Bug fix ('fix')

## Note to Reviewers

Using only your keyboard, attempt navigation to any of the entity creation links via the Create menu dropdown.

## Todo:

- [x] Re-implement arrow flip on open
- [x] SR verification
- [x] Cross-browser/device check
- [x] Cleanup

Jskobos commented on packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.tsx:
```diff
@@ -96,128 +139,99 @@ type CombinedProps = Props &
-          ))}
+          </MenuButton>
+          <MenuPopover className={classes.menuItemList} portal={false}>
+            <MenuItems className={classes.menuItemList}>
```
Do both of these need the same class?

WilkinsKa1 replied:
Good point, technically no, since I'm targeting styling with the data-reach attributes instead, the className is more of a wrapper now. I'll split these out!

johnwcallahan commented on packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenuItem.tsx:
```diff
@@ -118,23 +106,9 @@ class AddNewMenuItem extends React.Component<PropsWithStyles, State> {
-        )}
-      </MenuItem>
+      <div className={classes.root} data-qa-add-new-menu={title}>
+        {menuItemContent()}
```
I'm wondering why this content was in a function to begin with. I think you could just move the contents of the `menuItemContent` function into this return body.

WilkinsKa1 replied:
Good call!

johnwcallahan commented on packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenuItem.tsx:
```diff
@@ -71,19 +69,17 @@ const styles = (theme: Theme) =>
-  index: number;
-  count: number;
-}
+// interface Props extends MenuItems {
```
Any reason you want to keep this around? I vote to cut it.

johnwcallahan commented on packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.tsx:
```diff
@@ -96,128 +139,99 @@ type CombinedProps = Props &
+          <MenuButton
+            className={classes.button}
+            data-qa-add-new-menu-button
+            aria-label="Linode Create"
```
Not new with this PR, but I think this label is a little misleading.

johnwcallahan commented:
Oh, one other thing I noticed. If you open up dev tools (docked at the bottom of the screen), and expand them enough that the menu popover won't fit in the viewport, opening the Create menu takes you immediately to the Create Linode flow.

I observed the same thing on the [Reach UI documentation page](https://reacttraining.com/reach-ui/menu-button), so I wonder if it's either an issue with this component, or expected behavior for some reason.

WilkinsKa1 commented:
@Jskobos @DevDW @johnwcallahan Thanks for the reviews! I updated the contrast for focus/hover state and removed the animation (icon should still change colors on hover/focus as well though) 

John: I'm still looking into your last comment about the viewport callout! I can reproduce as well, but also unclear if its expected or not.

johnwcallahan commented on packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.tsx:
```diff
@@ -83,9 +135,7 @@ interface Props {
-interface State {
-  anchorEl?: HTMLElement;
-}
+interface State {}
```
This can be removed.
